### PR TITLE
Add DomainUpDown tests and remove dead code

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Add DomainUpDown tests and remove dead code

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2854)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs [hunk 1]
```diff
@@ -0,0 +1,1487 @@
...
+            Assert.Equal(userEdit, control.UserEdit);
+            Assert.True(control.IsHandleCreated);
+            Assert.Equal(0, invalidatedCallCount);
+            Assert.Equal(0, styleChangedCallCount);
+            Assert.Equal(0, createdCallCount);
+        }
+
+        [WinFormsTheory]
+        [MemberData(nameof(Sorted_WithItemsWithSelection_TestData))]
+        public void DomainUpDown_Sorted_SetWithItemsWithSelectionWithHandle_GetReturnsExpected(bool userEdit, bool value, string[] expectedItems, int expectedSelectedIndex)
```
Comment (JuditRose) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs]: ```value``` could be ```sorted``` for better readability.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs [hunk 2]
```diff
@@ -0,0 +1,1487 @@
...
+            Assert.Equal(userEdit, control.UserEdit);
+            Assert.True(control.IsHandleCreated);
+            Assert.Equal(0, invalidatedCallCount);
+            Assert.Equal(0, styleChangedCallCount);
+            Assert.Equal(0, createdCallCount);
+        }
+
+        [WinFormsTheory]
+        [CommonMemberData(nameof(CommonTestHelper.GetBoolTheoryData))]
+        public void DomainUpDown_Wrap_Set_GetReturnsExpected(bool value)
```
Comment (JuditRose) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs]: same here and everywhere else as well

File: src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs [hunk 3]
```diff
@@ -175,7 +175,7 @@ public object SelectedItem
                     //
                     for (int i = 0; i < Items.Count; i++)
                     {
-                        if (value != null && value.Equals(Items[i]))
+                        if (value.Equals(Items[i]))
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs]: I don't believe this is correct. It is possible to add `null` to the `Items` collection
image

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs]: Its alredy checked further up this method though, so this location is not reachable for value==null, no point checking it. Once you enable nullability analyzers they might even warn you that the check is redundant and value can't be null in this branch.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs]: I do prefer `value` as this is the implicit name of C# property setter parameters. Also good to be consistent across corefx/winforms tess

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DomainUpDownTests.cs]: See above

File: src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs]: Correct, we have the following code
```cs
// Treat null as selecting no item
//
if (value == null)
{
    SelectedIndex = -1;
}
else
{
    // Attempt to find the given item in the list of items
    //
    for (int i = 0; i < Items.Count; i++)
    {
        if (value.Equals(Items[i]))
        {
            SelectedIndex = i;
            break;
        }
    }
}
```

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs]: Thank you. I should have expanded the hidden section myself...

**Pull request merged**